tests/datadriven/test_sequenceAnnotations.py

```diff
@@ -150,14 +151,11 @@ def testGetFeatureById(self):
             feature.strand,
             self._testData["sampleStrand"])
 
+    @utils.ExpectException(exceptions.ObjectWithIdNotFoundException)
     def testGetFeatureFailsWithBadId(self):
         idString = str(self._testData["sampleFeatureId"]) + "W00t"
-        try:
-            compoundId = datamodel.FeatureCompoundId.parse(idString)
-            self._gaObject.getFeature(compoundId)
-            self.assertFalse("Exception should be thrown by this point")
-        except exceptions.ObjectWithIdNotFoundException:
-            pass
+        compoundId = datamodel.FeatureCompoundId.parse(idString)
```

Comment:
It didn't allow me to write a fancy decorator, that's what!

But I agree assertRaises is better because it scopes the line on which one expects the exception to be thrown, rather than the whole function.